### PR TITLE
Add NuGet authentication

### LINKED ISSUE #8404: Add authentication to NuGet config
Currently, the C# frontend NuGet config is using anonymous access to download the Nuget packages from artifactory.
While this works, it does not provide the Artifactory logs with enough information about the requests received (See [this thread for more info](https://discuss.sonarsource.com/t/disabling-anonymous-access-to-jfrog-artifactory-repox-1st-sprint-information-gathering)).

The appropriate credentials should be added to the NuGet config.

[Instructions to add credentials.](https://xtranet-sonarsource.atlassian.net/wiki/spaces/RE/pages/2972123145/JFrog+Artifactory+repox+authentication#NuGet) - we should use the Repox Artifactory NuGet token already present in Vault.

Related tickets:
- https://sonarsource.atlassian.net/browse/SONARSEC-4673
- https://sonarsource.atlassian.net/browse/BUILD-3140

### PR DESCRIPTION
**Pull request by sebastien-marichal**
Fixes #8404

This has been tested against a NuGet repository that requires authentication. (see 
BUILD-4368)
Until https://repox.jfrog.io/artifactory/api/nuget/nuget enforces authentication, it will work without authentication, issues may raise once authentication is enforced on this repository.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -70,7 +70,13 @@ stages:
       - task: NuGetToolInstaller@1
         displayName: "Install NuGet"
 
+      - script: 'dotnet nuget locals all --clear'
+        displayName: "NuGet Cache Clear"
```
Comment (sebastien-marichal) [inline azure-pipelines.yml:74]: This is to ensure there is no NuGet cache.
I will remove this change before merging.

File: analyzers/NuGet.Config [hunk 2]
```diff
@@ -1,9 +1,15 @@
-<?xml version="1.0" encoding="utf-8"?>
+﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
     <clear />
     <add key="Repox Artifactory (proxy)" value="https://repox.jfrog.io/artifactory/api/nuget/nuget" />
   </packageSources>
+  <packageSourceCredentials>
+    <Repox_x0020_Artifactory_x0020__x0028_proxy_x0029_>
```
Comment (andrei-epure-sonarsource) [inline analyzers/NuGet.Config:8]: this is so difficult to read, but it's actually `Repox Artifactory (proxy)` 
(hint for the next reviewer)

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

I'm waiting also for Malena to have a look before we merge it

File: analyzers/NuGet.Config (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/NuGet.Config:8]: What about renaming the key just to `Repox` to get rid of this problem?

File: docs/contributing-analyzer.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 All C# and VB.NET code analyzers present in SonarLint for Visual Studio, SonarQube and SonarCloud are being developed here. These analyzers rely on Roslyn 1.3.2 API.
 
+Before following any of the guides below, if you are external contributor you need to delete `sonar-dotnet\analyzers\NuGet.Config`.
```
Comment (pavel-mikula-sonarsource) [inline docs/contributing-analyzer.md:5]:
```suggestion
Before following any of the guides below, if you are external contributor you need to delete `analyzers\NuGet.Config`.
```

File: docs/contributing-analyzer.md [hunk 4]
```diff
@@ -2,6 +2,8 @@
...
+
 ## Working with the code
 
 1. Clone [this repository](https://github.com/SonarSource/sonar-dotnet.git)
```
Comment (pavel-mikula-sonarsource) [inline docs/contributing-analyzer.md:9]: CaYC: We don't use submodules anymore, you can delete the next line

Comment (pavel-mikula-sonarsource) [reply docs/contributing-analyzer.md:9]: I generally disagree with point 3 and the command prompt. But that's another PR

File: docs/contributing-analyzer.md [hunk 5]
```diff
@@ -51,6 +50,12 @@ Visual Studio 2022 version 17.6+ is required to build the project (due to source
...
         - %M2_HOME%\bin (`C:\Program Files\JetBrains\IntelliJ IDEA\plugins\maven\lib\maven3\bin` [Maven cli](https://maven.apache.org/install.html). Here installed via IntelliJ IDEA.)
         - SonarScanner for .NET folder and to the Scanner CLI ([SonarScanner download](https://github.com/SonarSource/sonar-scanner-msbuild/releases))
+    -  **Sonar internal only** These two steps require access to SonarSource internal resources and are not possible for external contributors
+        - **ORCHESTRATOR_CONFIG_URL** - url to `orchestrator.properties` file (for integration tests) in uri form (i.e. `file:///c:/something/orchestrator.properties`). See also: [Documentation in the orchestrator repository](https://github.com/sonarsource/orchestrator#configuration)
+        - **RULE_API_PATH** - path to folder containing the rule api jar. The rule api jar can be found at [repox.jfrog search](https://repox.jfrog.io/ui/artifactSearchResults?name=rule-api&type=artifacts) or [repox.jfrog private releases](https://repox.jfrog.io/ui/native/sonarsource-private-releases/com/sonarsource/rule-api/rule-api/)
+        - **ARTIFACTORY_URL** https://repox.jfrog.io/repox
```
Comment (pavel-mikula-sonarsource) [inline docs/contributing-analyzer.md:56]: What's this for?
```suggestion
```

File: docs/contributing-analyzer.md [hunk 6]
```diff
@@ -51,6 +50,12 @@ Visual Studio 2022 version 17.6+ is required to build the project (due to source
...
+        - **ARTIFACTORY_USER** your repox.jfrog username (see e.g. `orchestrator.properties`)
+        - **ARTIFACTORY_PASSWORD** the api key for repox.jfrog (see e.g. `orchestrator.properties`)
```
Comment (pavel-mikula-sonarsource) [inline docs/contributing-analyzer.md:58]: Is this really "api key" based on the latest changes? I think it's a token now.
```suggestion
        - **ARTIFACTORY_PASSWORD** the identity token for repox.jfrog (see e.g. `orchestrator.properties`)   
```

File: docs/contributing-analyzer.md [hunk 7]
```diff
@@ -154,11 +159,8 @@ After the debug session, remove the `Debugger.Launch()` line.
...
   * Open the root folder of the repo
   * Make sure the `its`, `sonar-csharp-plugin`, `sonar-dotnet-shared-library`, and `sonar-vbnet-plugin` folders are are imported as Maven modules (indicated by a blue square). Search for `pom.xml` in the folders and make it a maven project if not.
-* Add the following environment variables (*user* scope)
-  * **ARTIFACTORY_URL** https://repox.jfrog.io/repox
-  * **ARTIFACTORY_USER** your repox.jfrog username (see e.g. orchestrator.properties)
-  * **ARTIFACTORY_PASSWORD** the api key for repox.jfrog (see e.g. orchestrator.properties)
 * Create `settings.xml` in the `%USERPROFILE%\.m2` directory. A template can be found in the [Developer box section in the extranet](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DEV/pages/776711/Developer+Box#Maven-Settings). Change the username and password settings with the values from the environment variables above.
+* Make sure the orchestrator is configured [see here](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DEV/pages/776679/Integration+Tests#Configuration).
```
Comment (pavel-mikula-sonarsource) [inline docs/contributing-analyzer.md:163]: Why is this here? The information is redundant to the `ORCHESTRATOR_CONFIG_URL` environment variable
```suggestion
```

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Some polishing on the readme

File: analyzers/NuGet.Config (hunk 2 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/NuGet.Config]: This is done already in the first of the three last commits https://github.com/SonarSource/sonar-dotnet/pull/8514/commits/9543920a1328e5988fa6294b75be1ca5bd6ced35.

File: docs/contributing-analyzer.md (hunk 5 above)
Comment (mary-georgiou-sonarsource) [reply docs/contributing-analyzer.md:56]: 🤦

File: docs/contributing-analyzer.md (hunk 7 above)
Comment (mary-georgiou-sonarsource) [reply docs/contributing-analyzer.md:163]: True - and internals have no other choice than developing with an IDE.

File: docs/contributing-analyzer.md (hunk 6 above)
Comment (mary-georgiou-sonarsource) [reply docs/contributing-analyzer.md]: I deleted also (see e.g. `orchestrator.properties)` since the orchestrator properties get actually an API key.

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM
